File: content/code/count_word_occurrence_in_file_sol.py
import tempfile
import os

def test_count_word_occurrence_in_file():
    _, temporary_file_name = tempfile.mkstemp()
    with open(temporary_file_name, 'w') as f:
        f.write("one two one two three four")
    count = count_word_occurrence_in_file(temporary_file_name, "one")
    assert count == 2
    os.remove(temporary_file_name)

File: content/code/cpp/count_word_occurrence_in_file_sol.cpp
#include <filesystem> // for temp_directory_path(), requires C++17.
#include <fstream>

#include <catch2/catch.hpp>

#include "word_count.hpp"

TEST_CASE("Count occurrences of substring in file", "[count_word_occurrence_in_file]") {
  namespace fs = std::filesystem;
  auto tmp_dir{ fs::temp_directory_path() };
  auto fpath{ fs::temp_directory_path() / "temp_file" };

  std::ofstream s(fpath, std::ios::out | std::ios::trunc);
  s << "one two one two three four" << std::endl;
  s.close();

  REQUIRE(count_word_occurrence_in_file(fname, "one") == 2);
  REQUIRE(count_word_occurrence_in_file(fname, "three") == 1);
  REQUIRE(count_word_occurrence_in_file(fname, "six") == 0);

  fs::remove( fpath );
}
